File: swift/0424-longest-repeating-character-replacement.swift
class Solution {
    func characterReplacement(_ s: String, _ k: Int) -> Int {
        var count: [Character: Int] = [:]
        var strArray = Array(s)
        var result = 0
        var l = 0
        var maxF = 0

        for r in 0...s.count - 1 {
            count[strArray[r]] = count[strArray[r], default: 0] + 1
            maxF = max(maxF, count[strArray[r], default: 0])

            while (r - l + 1) - maxF > k {
                count[strArray[l]] = count[strArray[l], default: 0] - 1
                l += 1 
            }
            result = max(result, r - l + 1)
        }
        return result
    }
}

File: python/0424-longest-repeating-character-replacement.py
from typing import List, Optional
import collections
from utils import TreeNode, SLLNode as ListNode

class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        
        l = 0
        maxf = 0
        for r in range(len(s)):
            count[s[r]] = 1 + count.get(s[r], 0)
            maxf = max(maxf, count[s[r]])

            if (r - l + 1) - maxf > k:
                count[s[l]] -= 1
                l += 1

        return (r - l + 1)
